crates/ra_hir/src/ty/tests.rs

```diff
@@ -287,6 +287,32 @@ fn test(x: &str, y: isize) {
     );
 }
 
+#[test]
+fn infer_array() {
+    check_inference(
+        r#"
+fn test(x: &str, y: isize) {
+    let a = [x];
+    let b = [a, a];
+    let c = [b, b];
+
+    let d = [y, 1, 2, 3]
```

Comment:
there should be semicolons here and in the line below... it's a testament to the parser's error recovery that this still works :)